Remove Batch tests; moved to standalone Batch TCK: https://github.com…

scottkurz opened the pull request:
…/eclipse-ee4j/batch-tck

Signed-off-by: Scott Kurz <skurz@us.ibm.com>

---
name: Pull Request
about: Create a pull request for a Platform TCK change
title: ''
labels: ''
assignees: ''

---

**Fixes Issue**
Fixes #574 

**Related Issue(s)**
Specify any related issue(s) links.

**Describe the change**
Remove the Batch tests now covered via standalone TCK: 
https://github.com/eclipse-ee4j/batch-tck

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @scottmarlow

---

I think we are very close to adding a path covering the standalone platform tests in each of an EJB + Servlet context, as agreed on.   We're not quite there so I left this in 'draft' state.. but this is just an FYI that we're close.

gurunrao commented:
@scottmarlow - If we go ahead with PR, we will not be able to run batch tests with Platform TCK EJB vehicle - https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/install/jakartaee/other/vehicle.properties#L264

scottkurz commented:
> @scottmarlow - If we go ahead with PR, we will not be able to run batch tests with Platform TCK EJB vehicle - https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/install/jakartaee/other/vehicle.properties#L264

We expect to add an EJB path (as a variation to the existing Servlet path) within the platform portion of the standalone Batch TCK, implemented via Arquillian.    Tracked via: https://github.com/eclipse-ee4j/batch-tck/issues/22

scottkurz commented:
@scottmarlow  - I think we should also add some small amount of documentation that says:  "To run the Platform, go run the platform portion of the Jakarta Batch  standalone TCK".   

Is there another example of where this is already done?     If not, I can take a shot at it?  Or would it be doc'd outside of this repo and more at some other, process level somewhere?

scottmarlow commented:
> > @scottmarlow - If we go ahead with PR, we will not be able to run batch tests with Platform TCK EJB vehicle - https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/install/jakartaee/other/vehicle.properties#L264
> 
> We expect to add an EJB path (as a variation to the existing Servlet path) within the platform portion of the standalone Batch TCK, implemented via Arquillian. Tracked via: [eclipse-ee4j/batch-tck#22](https://github.com/eclipse-ee4j/batch-tck/issues/22)

:+1: 

IMO, we shouldn't remove the Batch tests from the Platform TCK until the work is complete to include an EJB path.  Thank you Scott for creating this pr for dealing the test removal, very much appreciated!

scottmarlow commented:
> @scottmarlow - I think we should also add some small amount of documentation that says: "To run the Platform, go run the platform portion of the Jakarta Batch standalone TCK".

:+1: that sounds perfect to me.  We can always add more details about the Batch standalone TCK (e.g. link to the Batch TCK or perhaps to https://jakarta.ee/specifications/batch/2.1/ page which will later have the TCK link).

> 
> Is there another example of where this is already done? If not, I can take a shot at it? Or would it be doc'd outside of this repo and more at some other, process level somewhere?

Yes, https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/user_guides/jakartaee/src/main/jbake/content/intro.adoc contains a few relevant sections.

About what Platform TCK tests:
```
Technologies Tested with Jakarta EE 9.1 Platform TCK

The Jakarta EE 9.1 Platform TCK test suite includes compatibility tests for the
following required and optional Jakarta EE 9.1 technologies:

* Jakarta Enterprise Beans 4.0 +
* Jakarta Servlet 5.0
* Jakarta Server Pages 3.0
* Jakarta Expression Language 4.0
* Jakarta Messaging 3.0
* Jakarta Transactions 2.0
* Jakarta Mail 2.0
* Jakarta Connectors 2.0
* Jakarta Enterprise Web Services 2.0 (optional)
* Jakarta RESTful Web Services 3.0
* Jakarta WebSocket 2.0
* Jakarta JSON Processing 2.0
* Jakarta JSON Binding 2.0
* Jakarta Concurrency 2.0
* Jakarta Batch 2.0
* Jakarta Authorization 2.0
* Jakarta Authentication 2.0
* Jakarta Standard Tag Library 2.0
* Jakarta Faces 3.0
* Jakarta Security 2.0
* Jakarta Annotations 2.0
* Jakarta Persistence 3.0
* Jakarta Bean Validation 3.0
* Jakarta Managed Beans 2.0
* Jakarta Interceptors 2.0
* Jakarta Contexts and Dependency Injection 3.0
* Jakarta Dependency Injection 2.0
* Jakarta Debugging Support for Other Languages 2.0
* Jakarta Enterprise Beans 3.2 and earlier entity beans and associated Jakarta Enterprise Beans QL (optional)
* Jakarta Enterprise Beans 2.x API group (optional)
* Jakarta Enterprise Web Services 2.0 (optional)
* Jakarta SOAP with Attachments 2.0 (optional)
* Jakarta Web Services Metadata 3.0 (optional)
* Jakarta XML Web Services 3.0 (optional)
```

Additional Jakarta EE 9.1 Platform TCK Requirements:
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In addition to the instructions and requirements described in this
document, all Jakarta EE 9.1 and Jakarta EE 9.1 Web Profile implementations must
also pass the standalone TCKs for the following technologies:

* Jakarta Contexts and Dependency Injection 3.0
* Jakarta Dependency Injection 2.0
* Jakarta Bean Validation 3.0
* Jakarta XML Binding 3.0

For more information about the Jakarta Contexts and Dependency Injection technology, 
see the specification at `https://jakarta.ee/specifications/cdi/3.0/`

For more information about the Jakarta Dependency Injection,
see the specification at `https://jakarta.ee/specifications/dependency-injection/2.0/` 

For more information about the Jakarta Bean Validation technology, see the
specification at `https://jakarta.ee/specifications/bean-validation/3.0/`

For more information about the Jakarta XML Binding technology, see the
specification at `https://jakarta.ee/specifications/xml-binding/3.0/`

[[GBFQW]][[getting-started-with-the-jakarta-ee-9-platform-tck-test-suite]]
```

scottmarlow commented:
> @scottmarlow marking for review, though I'd imagine I didn't get everything.
> 
> This is complicated by the fact that "batch" as in generic grouping is used in a number of places that are NOT referring to Jakarta Batch, preventing a simple find/replace.

Agreed.

> # Batch Standalone TCK ready
> 
> First, with the merge of: [eclipse-ee4j/batch-tck#40](https://github.com/eclipse-ee4j/batch-tck/pull/40) and the pending related doc PR: [eclipse-ee4j/batch-tck#41](https://github.com/eclipse-ee4j/batch-tck/pull/41) we do now have functional equivalence in that we are testing from both a servlet and Enterprise Bean context
> # Removed from platform TCK
> 
> As you can see I removed
> 
>     1. the Batch TCK tests themselves
> 
>     2. the Batch signature files
> 
>     3. references (that I saw) to batch/jbatch in the automation
> 
>     4. references (that I saw) to batch/jbatch in the doc
> 
> 

:+1: 

> ## Comment
> 
> I updated the entry in: user_guides/jakartaee/src/main/jbake/content/intro.adoc as mentioned but I noticed it's still at the EE 9.1 level of spec versions. There isn't some other branch I should be working off is there?
> # Question
> 
> One thing I was unsure about is whether to remove signature test related entries. We do have a full set of signature tests in the Batch Standalone TCK, so maybe we don't need them here. OTOH, I see CDI-related entries and CDI has its own sig tests, IIUC, so maybe they should be left in?

We will try to avoid duplication of tests that users run so I agree with removing the Jakarta Batch TCK signature test from the Platform TCK (as long as the Standalone Batch TCK signature map file is equivalent to the Jakarta Platform TCK Batch TCK signature map file).  

If for example you notice that the Batch TCK signature map file is missing some classes for a certain Jakarta EE SPEC API, that likely can be resolved by changing how you generate the signature map file to include the class package name for the incomplete SPEC API (for example of that type of issue being addressed, see https://github.com/eclipse-ee4j/cdi-tck/pull/348.)  

Note that the https://github.com/eclipse-ee4j/jakartaee-tck/tree/master/user_guides/jakartaee/src/main/jbake/content/intro.adoc is in need of an update to section 1.5 (`Additional Jakarta EE 9.1 Platform TCK Requirements`) but that doesn't need to change with this pull request (a lot of versions and text is pending to be updated in the Platform TCK user guide).  

@scottkurz could you please send email to the Platform, Platform TCK, Batch mailing lists asking for a review of this pull request for removal of duplicate Jakarta Batch tests?  Or ping me back if you would prefer that I send that, either way is fine.  

https://github.com/eclipse-ee4j/jakartaee-platform/wiki/Guidelines_For_Standalone_TCKs mentions that we want the different teams to have a chance to review changes that remove Platform TCK tests due to duplication between Standalone TCK + Platform TCK.

Thanks!
Scott

scottkurz commented:
OK, updated PR removing the signature tests.  This is to the best of my ability as I can determine just by eyeballing the source;  I'm not verifying this by actually running anything. (I haven't done this previously or I would have tried.)   

I think I deleted all references to Jakarta Batch  and left all references to "batch testing" and "batch" APIs like JDBC "batch update".   But please check.

## QUESTION?
One place in particular I wasn't sure about was:  **user_guides/jakartaee/src/main/jbake/content/rules.adoc**  
I left it in since it seemed to be saying Batch was part of the Platform in a more abstract sense.  But maybe I'm wrong and this refers to a more specific test construct and so Jakarta Batch  should also be removed from this list.

scottkurz commented:
@scottmarlow just noting we made it to the two week mark.  I don't see that there have been any comments. 

I have been rebasing/fixing the merge conflicts.   As I mentioned I'm not familiar enough to go ahead and try to test these changes (i.e. run the TCK w/o Batch).

scottmarlow commented on docker/build_signatures.sh:
```diff
@@ -58,15 +58,14 @@ export deliverabledir=signature-repository
 export tckname=jakartaee
 
-export INCLUDES="jakarta.activation:jakarta.annotation:javax.annotation.processing:jakarta.annotation.security:jakarta.annotation.sql:jakarta.batch.api:jakarta.batch.api.chunk:jakarta.batch.api.chunk.listener:jakarta.batch.api.listener:jakarta.batch.api.partition:jakarta.batch.operations:jakarta.batch.runtime:jakarta.batch.runtime.context:jakarta.decorator:jakarta.ejb:jakarta.ejb.embeddable:jakar ...
+export INCLUDES="jakarta.activation:jakarta.annotation:javax.annotation.processing:jakarta.annotation.security:jakarta.annotation.sql:jakarta.decorator:jakarta.ejb:jakarta.ejb.embeddable:jakarta.ejb.spi:jakarta.el:jakarta.enterprise.concurrent:jakarta.enterprise.context:jakarta.enterprise.context.control:jakarta.enterprise.context.spi:jakarta.enterprise.event:jakarta.enterprise.inject:jakarta.ent ...
```
We should not update docker/build_signatures.sh as per my general comment about having Batch SPEC API signature checking in both Standalone + Platform TCK.

scottmarlow commented on docker/pom.xml:
```diff
@@ -52,8 +52,6 @@
         <jakarta.transaction-api.version>2.0.1-RC1</jakarta.transaction-api.version>
         <!-- Jakarta Connectors -->
         <jakarta.resource-api.version>2.1.0</jakarta.resource-api.version>
-        <!-- Jakarta Batch -->
```
We should leave docker/pom.xml as is also as that is used for generating signature map files.

scottmarlow commented on src/com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java:
```diff
@@ -85,14 +85,14 @@
   public static final String KEYWORD_JAVAEE = "ejb interceptors caj jms wsmd javamail"
       + " cdi di beanval persistence jaxb saaj jaxws connector"
-      + " jacc jaspic jsonp jta el servlet jsf jaxrs websocket batch concurrency jsonb securityapi";
+      + " jacc jaspic jsonp jta el servlet jsf jaxrs websocket concurrency jsonb securityapi";
```
We should keep current src/com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java settings also for signature testing.

scottmarlow commented on src/com/sun/ts/tests/signaturetest/recordSigs.sh.txt:
```diff
@@ -62,7 +62,7 @@ s1as_modules="${AS_HOME}/modules"
 # mac terminal.  The safest fix seemed to be to rename it....so we did.
 
-sigTestClasspath=${s1as_modules}/javax.json.jar${pathsep}${s1as_modules}/javax.batch-api.jar${pathsep}${s1as_modules}/javax.interceptor-api.jar${pathsep}${s1as_modules}/stax2-api.jar${pathsep}${s1as_modules}/javax.enterprise.concurrent-api.jar${pathsep}${s1as_modules}/jakarta.websocket-api.jar${pathsep}${s1as_modules}/javax.jms-api.jar${pathsep}${s1as_modules}/javax.xml.registry-api.jar${pathsep} ...
+sigTestClasspath=${pathsep}${s1as_modules}/javax.interceptor-api.jar${pathsep}${s1as_modules}/stax2-api.jar${pathsep}${s1as_modules}/javax.enterprise.concurrent-api.jar${pathsep}${s1as_modules}/jakarta.websocket-api.jar${pathsep}${s1as_modules}/javax.jms-api.jar${pathsep}${s1as_modules}/javax.xml.registry-api.jar${pathsep}${s1as_modules}/jakarta.faces.jar${pathsep}${s1as_modules}/bean-validator.j ...
```
I think this file is ignored, so either keep or revert the change to this file.

scottmarlow commented on install/jakartaee/bin/sig-test-pkg-list.txt:
```diff
@@ -37,14 +37,6 @@ jakarta.annotation
 jakarta.annotation.security
 jakarta.annotation.sql
-jakarta.batch.api
-jakarta.batch.api.chunk
```
It shouldn't matter if you keep or remove this change for signature testing as we are using docker/build_signatures.sh

scottmarlow commented on install/jakartaee/bin/sig-test.map:
```diff
@@ -157,10 +157,6 @@ jakarta.json.bind=2.1
 # (This is JSR-356)
 jakarta.websocket=2.1
 
-# Batch 2.1 
```
We should keep this change for running the signature testing with Platform TCK.

scottmarlow reviewed (changes requested):
Looks good to me other than the requested changes requested above.

scottkurz commented:
@scottmarlow I updated this PR adding back the signature test removals.  Thanks for the detailed comments.  I think I captured them correctly.   

In case you're trying to re-review, let me note that I left the first 3 commits in the PR and replaced only the 4th commit.

If you want to simply diff the new, updated PR (with the new 4th commit) with what I had before (but force pushed over) you can see the earlier copy at: 
https://github.com/scottkurz/jakartaee-tck/tree/remove-batch-tck-prev

I didn't simply revert the 4th commit because I had a couple things besides sigtest stuff in there.

Thanks.

scottmarlow commented:
I'll do some testing on Jenkins of this change this week.

scottkurz commented:
Thx @scottmarlow  - just reminding you I'm not a committer able to merge the PR myself.

scottmarlow commented:
> Thx @scottmarlow - just reminding you I'm not a committer able to merge the PR myself.

Thanks, just giving others a change to review also before merging.

gurunrao commented:
@scottkurz @scottmarlow  - trying to get test run of platform TCK with the PR changes, blocked by Platform TCK build issues.

scottmarlow commented:
Deferring the merge until after we fix Platform TCK build

scottmarlow commented:
Started https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck-scottmarlow/job/remove-batch-moved-to-standalone-tck/2/ to verify that we can still build the Platform TCK after merging this change.
